src/plugins/intel_gpu/src/graph/impls/common/condition.cpp

```diff
@@ -41,9 +41,14 @@ struct condition_impl : typed_primitive_impl<condition> {
         auto pred = condition_inst::get_pred_from_memory(instance.pred_memory_ptr(), stream);
         network::ptr executed_net = pred ? instance.get_net_true() : instance.get_net_false();
         auto branch = pred ? instance.get_branch_true() : instance.get_branch_false();
-        executed_net->set_shape_predictor(instance.get_network().get_shape_predictor());
+        bool can_skip_subgraph = branch.inner_program->can_be_optimized();
+        if (can_skip_subgraph)
```

Comment:
Was your intention not to set shape predictor for networks which can be optimized?
```suggestion
        if (!can_skip_subgraph)
```

Reply:
that was typo! thanks!